Lib/Src/CGpioWrapper.cpp

```diff
@@ -0,0 +1,64 @@
+/*
+ * CGpioWrapper.cpp
+ *
+ *  Created on: 14 Feb 2022
+ *      Author: salavat.magazov
+ */
+
+#include "CGpioWrapper.h"
+
+/**
+ * @brief Create instance of a GPIO pin wrapper.
+ *
+ * @param p_port Pointer to GPIO port
+ * @param pin Pin number
+ */
+CGpioWrapper::CGpioWrapper(GPIO_TypeDef *p_port, uint32_t pin)
+    : mp_port(p_port),
+      m_pin(pin)
+{
+    if (mp_port == NULL)
+    {
+        Error_Handler();
+    }
+    if (!IS_GPIO_PIN(m_pin))
```

Comment:
If you use ```uint16_t``` I think this becomes redundant. Here's the IS_GPIO_PIN macro, let me know if you agree. 
```
#define GPIO_PIN_MASK              (0x0000FFFFU) /* PIN mask for assert test */
#define IS_GPIO_PIN(__PIN__)        ((((uint32_t)(__PIN__) & GPIO_PIN_MASK) != 0x00U) &&\
                                     (((uint32_t)(__PIN__) & ~GPIO_PIN_MASK) == 0x00U))
```

Reply:
Just realised it's still useful to check the pin is not 0x00.

Reply:
It also checks that only one bit is set in a mask.